src/DDTrace/Integrations/PDO.php

```diff
@@ -0,0 +1,230 @@
+<?php
+
+namespace DDTrace\Integrations;
+
+use DDTrace\Tags;
+use DDTrace\Types;
+use OpenTracing\GlobalTracer;
+
+class PDO
+{
+    private static $connections = [];
+    private static $statements = [];
+
+    /**
+     * Static method to add instrumentation to PDO requests
+     */
+    public static function load()
+    {
+        if (!extension_loaded('ddtrace')) {
+            trigger_error('ddtrace extension required to load PDO integration.', E_USER_WARNING);
+            return;
+        }
+
+        // public PDO::__construct ( string $dsn [, string $username [, string $passwd [, array $options ]]] )
+        dd_trace('PDO', '__construct', function (...$args) {
+            $scope = GlobalTracer::get()->startActiveSpan('PDO.__construct');
+            $span = $scope->getSpan();
+            $span->setTag(Tags\SPAN_TYPE, Types\SQL);
+            $span->setTag(Tags\SERVICE_NAME, 'PDO');
+            $span->setResource('PDO.__construct');
+
+            try {
+                $this->__construct(...$args);
+                PDO::storeConnectionParams($this, $args);
+                return $this;
+            } catch (\Exception $e) {
+                $span->setError($e);
+                throw $e;
+            } finally {
+                $scope->close();
+            }
+        });
+
+        // public int PDO::exec(string $query)
+        dd_trace('PDO', 'exec', function ($statement) {
+            $scope = GlobalTracer::get()->startActiveSpan('PDO.exec');
+            $span = $scope->getSpan();
+            $span->setTag(Tags\SPAN_TYPE, Types\SQL);
+            $span->setTag(Tags\SERVICE_NAME, 'PDO');
+            $span->setResource($statement);
+            PDO::setConnectionTags($this, $span);
+
+            try {
+                $result = $this->exec($statement);
+                $span->setTag('db.rowcount', $result);
+                return $result;
+            } catch (\Exception $e) {
+                $span->setError($e);
+                throw $e;
+            } finally {
+                $scope->close();
+            }
+        });
+
+        // public PDOStatement PDO::query(string $query)
+        // public PDOStatement PDO::query(string $query, int PDO::FETCH_COLUMN, int $colno)
+        // public PDOStatement PDO::query(string $query, int PDO::FETCH_CLASS, string $classname, array $ctorargs)
+        // public PDOStatement PDO::query(string $query, int PDO::FETCH_INFO, object $object)
+        // public int PDO::exec(string $query)
+        dd_trace('PDO', 'query', function (...$args) {
+            $scope = GlobalTracer::get()->startActiveSpan('PDO.query');
+            $span = $scope->getSpan();
+            $span->setTag(Tags\SPAN_TYPE, Types\SQL);
+            $span->setTag(Tags\SERVICE_NAME, 'PDO');
+            $span->setResource($args[0]);
+            PDO::setConnectionTags($this, $span);
+
+            try {
+                $result = $this->query(...$args[0]);
```

Comment:
I think that here what we want to do is `$result = $this->query(...$args)` instead of only the first arg.

Otherwise we are not passing all the arguments:

```
function wrapper(...$args) {
    print_r("wrapper\n");
    print_r($args);
    query(...$args[0]);
}

function query(...$args) {
    print_r("query\n");
    print_r($args);
}

wrapper(1, 2, 3);

```

prints

```
wrapper
Array
(
    [0] => 1
    [1] => 2
    [2] => 3
)
PHP Warning:  Only arrays and Traversables can be unpacked in /Users/luca.abbati/projects/datadog/dd-trace-php/playground.php on line 7

Warning: Only arrays and Traversables can be unpacked in /Users/luca.abbati/projects/datadog/dd-trace-php/playground.php on line 7
query
Array
(
)

```

Reply:
Good catch, thanks!